ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js

```diff
@@ -112,8 +112,13 @@
         }
 
         updateValidity(validity) {
-            const valid = validity.valid ? validity.valid : false;
             let widgets = this.widget;
+            if(validity.valid === undefined) {
+                this.element.removeAttribute(FormView.Constants.DATA_ATTRIBUTE_VALID);
+                widgets.forEach(widget => widget.removeAttribute(FormView.Constants.ARIA_INVALID));
+                return;
+            }
+            const valid = validity.valid;
             this.element.setAttribute(FormView.Constants.DATA_ATTRIBUTE_VALID, valid);
```

Comment:
Can we try to re-use the code ? Let's wrap setting/removing these aria and data attributes in a function and re-use it in both if/else statement